starlark/eval_test.go

```diff
@@ -540,12 +546,66 @@ Error: join: in list, want string, got int`,
 	} {
 		globals := starlark.StringDict{"i": starlark.MakeInt(i)}
 		_, err := starlark.ExecFile(thread, "crash.star", src2, globals)
-		if got := getBacktrace(err); got != want {
+		if got := getBacktrace(t, err); got != want {
 			t.Errorf("error was %s, want %s", got, want)
 		}
 	}
 }
 
+func TestLoadBacktrace(t *testing.T) {
+	// This test ensures that load() does NOT preserve stack traces,
+	// but that API callers can get them with Unwrap().
+	// For discussion, see:
+	// https://github.com/google/starlark-go/pull/244
+	const src = `
+load('crash.star', 'x')
+`
+	const loadedSrc = `
+def f(x):
+  return 1 // x
+
+f(0)
+`
+	thread := new(starlark.Thread)
+	thread.Load = func(t *starlark.Thread, module string) (starlark.StringDict, error) {
+		return starlark.ExecFile(new(starlark.Thread), module, loadedSrc, nil)
+	}
+	_, err := starlark.ExecFile(thread, "root.star", src, nil)
+
+	const want = `Traceback (most recent call last):
+  root.star:2:1: in <toplevel>
+Error: cannot load crash.star: floored division by zero`
+	if got := getBacktrace(t, err); got != want {
+		t.Errorf("error was %s, want %s", got, want)
+	}
+
+	unwrapEvalError := func(err error) *starlark.EvalError {
+		var result *starlark.EvalError
+		for {
+			if evalErr, ok := err.(*starlark.EvalError); ok {
+				result = evalErr
+			}
+
+			wrapper, isWrapper := err.(Wrapper)
```

Comment:
// TODO: use errors.Unwrap when go >=1.13 is everywhere.

Reply:
done